fellowship.js

```diff
@@ -18,26 +18,80 @@ var fellowship = {
   ],
   makeMiddleEarth: function(lands) {
     console.log(lands)
+
     // create a section tag with an id of middle-earth
-    // add each land as an article tag
-    // inside each article tag include an h1 with the name of the land
+    var midEarthSect = document.createElement("section");
+    midEarthSect.setAttribute("id", "middle-earth");
+
+    for (var land_index in lands) {
+      // add each land as an article tag
+      var landArticle = document.createElement("article");
+
+      // setting an id for the article so we can fine the land easily
+      landArticle.setAttribute("id", lands[land_index]);
+
+      // inside each article tag include an h1 with the name of the land
+      var landHeader = document.createElement("h1");
+
+      // get the land name using index
+      landHeader.innerHTML = lands[land_index];
+
+      // add these as children
+      landArticle.appendChild(landHeader);
+      midEarthSect.appendChild(landArticle);
+    }
+
     // append middle-earth to your document body
+    document.body.appendChild(midEarthSect);
   },
   makeHobbits: function(hobbits) {
+    var ulHobbitsInTheShire = document.createElement("ul");
+
+    for (var hobbit_index in hobbits)
+    {
+      var liHobbit = document.createElement("li");
+      var classTag = document.createElement("div");
+      classTag.setAttribute("id", hobbits[hobbit_index]);
+      classTag.setAttribute("class", "hobbit");
+      classTag.innerHTML = hobbits[hobbit_index];
+      liHobbit.appendChild(classTag);
+      ulHobbitsInTheShire.appendChild(liHobbit);
+    }
+    document.body.appendChild(ulHobbitsInTheShire);
     // display an unordered list of hobbits in the shire
     // give each hobbit a class of hobbit
   },
   keepItSecretKeepItSafe: function() {
+
     // create a div with an id of 'the-ring'
+    var classTag = document.createElement("div");
+    classTag.setAttribute("id", "the-ring");
+
     // add the ring as a child of Frodo
+    document.getElementById("Frodo Baggins").appendChild(classTag);
   },
   makeBuddies: function(buddies) {
     // create an aside tag
+    var asideTag = document.createElement("aside");
+    var ulOfBuddies = document.createElement("ul");
+
+    for (var buddy_index in buddies)
+    {
```

Comment:
Where possible, it's best to follow conventional styles in terms of using new lines, etc. In this case, the standard formatting for loops would look like:

```js
for (var buddy_index in buddies) {
  // some code
}